Source/WebCore/rendering/svg/RenderSVGText.cpp

```diff
@@ -548,6 +537,15 @@ void RenderSVGText::paint(PaintInfo& paintInfo, const LayoutPoint& paintOffset)
     UNUSED_PARAM(paintOffset);
 #endif
 
+    if (paintInfo.context().paintingDisabled())
+        return;
+
+    if (paintInfo.phase != PaintPhase::ClippingMask && paintInfo.phase != PaintPhase::Mask && paintInfo.phase != PaintPhase::Foreground && paintInfo.phase != PaintPhase::Outline && paintInfo.phase != PaintPhase::SelfOutline)
+        return;
+
+    if (!paintInfo.shouldPaintWithinRoot(*this))
+        return;
```

Comment:
These checks are already done for the LBSE code path I think? If so then it may be nicer to put them in the #else section.